InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -0,0 +1,202 @@
+/***********************************************************************************************************************
+**
+** Copyright (c) 2011, 2015 ETH Zurich
+** All rights reserved.
+**
+** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+** following conditions are met:
+**
+**    * Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+**      disclaimer.
+**    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+**      following disclaimer in the documentation and/or other materials provided with the distribution.
+**    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+**      derived from this software without specific prior written permission.
+**
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+***********************************************************************************************************************/
+
+#include "CompositeQuery.h"
+
+#include "../graph/Graph.h"
+
+namespace InformationScripting {
+
+CompositeQuery::~CompositeQuery()
+{
+	SAFE_DELETE(inNode_);
+	for (auto node : nodes_) SAFE_DELETE(node);
+	// Don't delete the outNode_ as that we delete all Graphs which we returned in the execute methods.
+}
+
+QList<Graph*> CompositeQuery::execute(QList<Graph*> input)
+{
+	inNode_->calculatedOutputs_ = input;
+	// Nodes for which we have all dependencies calculated:
+	QQueue<QueryNode*> workingSetNodes;
```

Comment:
Call this something more descriptive. It seems like you put here things that have been just executed, so why not:
`justExecuted`.

Then you can also rename the set below from`visitedNodes` to `fullyProcessedQueries` or something else that is more descriptive than the current one.